generator/src/main/java/org/corfudb/generator/LongevityApp.java

```diff
@@ -61,19 +61,27 @@ public LongevityApp(long durationMs, int numberThreads, String configurationStri
     }
 
     /**
-     * Let the workers finish naturally (thanks to the timer) and then kill
+     * Give a chance to the workers to finish naturally (thanks to the timer) and then kill
      * the producer and the checkpointer.
      *
      * At the end of the duration, we give some margin for the workers to finish their task before shutting
      * down the thread pool.
-     * If the application is hung (which can happen when something goes wrong) we will still terminate.
+     *
+     * If the application is hung (which can happen when something goes wrong) we do a hard exit. If any thread
+     * is not playing nice with Interrupted exceptions (which can be a bug as well), this is the only way to
+     * be sure we terminate.
      */
     private void waitForAppToFinish() {
         workers.shutdown();
         try {
-            workers.awaitTermination(durationMs + APPLICATION_TIMEOUT_IN_MS, TimeUnit.MILLISECONDS);
+            boolean finishedInTime = workers.
+                    awaitTermination(durationMs + APPLICATION_TIMEOUT_IN_MS, TimeUnit.MILLISECONDS);
+            String livenessState = finishedInTime ? "Success" : "Fail";
+            Correctness.recordOperation("Liveness, " + livenessState, false);
+            if (!finishedInTime) {
+                System.exit(0);
```

Comment:
Shouldn't you return an error code in this case?

Reply:
Sure. Fixed.